### PR TITLE
Allowing extra headers on a per email basis

### PR DESCRIPTION
**Pull request by stkem**
For reply-to and various headers related to spam filter traversal.
@andrewconner @eishay @leogrim could one of you give this a quick review? (Being the three people with the most commits on the email related code)

### CODE REVIEW

**Comment by andrewconner** [comment]
Do we use these, or it's just for the future?

**Comment by leogrim** [comment]
lgtm

File: kifi-backend/modules/shoebox/app/com/keepit/common/mail/SendgridMailProvider.scala [hunk 1]
```diff
@@ -31,6 +31,8 @@ class SendgridMailProvider @Inject() (
     heimdal: HeimdalServiceClient)
   extends MailProvider with Logging {
 
+  private val validHeaders = Seq[String]("Reply-To", "List-Unsubscribe", "Precedence")
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/common/mail/SendgridMailProvider.scala]: May be nice to have type safety here. trait / case class combo, or object / val combo. Not particularly easy, at email creation site, to remember it's "List-Unsubscribe"

Comment (stkem) [inline kifi-backend/modules/shoebox/app/com/keepit/common/mail/SendgridMailProvider.scala]: Good idea. Adding.

**Comment by stkem** [comment]
@andrewconner we'll be using it shortly (i.e pre-launch)

**Pull request merged**